### DISCUSSION TITLE
hello. I am getting this error in the following js 13: Error: Unknown root exit status. I was wondering if any of you seniors have had this experience?

### DISCUSSION THREAD

**Discussion by jungjh565** [vercel/next.js#66485] [Help]
### Summary

hello.
I am getting this error in the following js 13: Error: Unknown root exit status.

I have the code below, if (typeof window !== ‘undefined’) I apply it, I get Error: Unknown root exit status.  I get this error and if I don't apply it, there is no error, but when I view the source, the file is in error. How can I deal with this situation?
‘use client’;
import { useCallback, useEffect, useLayoutEffect, useState } from ‘react’;

function useMediaQuery() {
  const [matches, setMatches] = useState(false);

  const handleCheckFuc = useCallback(() => {
    if (typeof window !== ‘undefined’) {
      const mediaQuery = window.matchMedia(‘(max-width: 767px)’);
      setMatches(mediaQuery.matches);
    }
  }, []);

  useEffect(() => {
    handleCheckFuc();
  }, [handleCheckFuc]);

  return { isMobile: () => matches };
}

export default useMediaQuery;

**Comment by icyJoseph** [comment]
Where are you using this hook? in a server component? It won't work if used directly on a server component, it has to be used by a client component.

**Reply by jungjh565** [reply]
Thank you for your reply. I'm using it in the client component.

**Reply by icyJoseph** [reply]
Aha, are you maybe passing `isMobile` as a prop.

**Reply by jungjh565** [reply]
Aha, are you saying that I should call it from the component and use it instead of using the isMobile prop?

**Reply by icyJoseph** [reply]
No I am not sure, show us how you use this hook.

**Reply by jungjh565** [reply]
```
const { isMobile } = useMediaQuery();

const handleAppBtn = React.useCallback(() => {
if (isMobile()) {}
},[isMobile])
```
```
 <MobileToggleMenu
           checkMobile={isMobile() as boolean}   />
```
```
 <BasicBtn width={isMobile() ? "100%" : "fit-content"} />
```

This is how I'm using it. 

We're using it inside a function, and we're using it down to a prop.
The fetch is const { isMobile } = useMediaQuery(); so it's being used inside ‘use client’.

**Reply by icyJoseph** [reply]
Ok, so I think this causes an issue where, server starts false, and then you poorly synchronize within the client, so if the width is larger than 767px, you switch to true, during render, and crash out. 

If you narrow down your viewport, to less than 767px, it probably won't crash. For this kind of thing it is better to use `CSS` to the styling, rather than JavaScript, as it will necessarily cause divergence.

- https://github.com/reactwg/server-components/discussions/4

Do remember, that client components are SSR'd too.

**Reply by jungjh565** [reply]
thank you I think I know something

**Reply by vogelino** [reply]
@jungjh565 What did you discover? Anyone found an explanation as to why the `Unknown root exit status` error is thrown? What does it mean? When does this happen?

**Comment by JClackett** [comment]
Yeah im getting this error quite often, im embedding a page in an iframe and after a server action am having to redirect, but sometimes get this root exit status error

**Reply by JClackett** [reply]
Also getting this on just a plain nextjs route, literally just a login form on it calling a server action.

**Comment by umang0620** [comment]
@JClackett @vogelino @icyJoseph is this error is safe to ignore in sentry?

getting same error from below file, and i am not able to track from where it's coming!
<img width="1271" height="235" alt="image" src="https://github.com/user-attachments/assets/219fbe1f-81f4-48d4-abdf-6755db88f475" />

**Reply by icyJoseph** [reply]
No idea. Never got to even be able to reproduce it. I found this blog post: https://blacksheepcode.com/posts/migrating_from_remix_to_next

Where they have `5.2 Mysterious Minified React error #329 / Unknown root exit status error.` - and seem to attribute it to Cypress tests? 🤷 I don't have time right now to clone the repo and try to verify if it still happens there.

**Reply by umang0620** [reply]
It seems that there is already an open issue for this.

Issue:- https://github.com/vercel/next.js/issues/56441

I just want to confirm that this error doesn’t have any end-user impact and can be safely ignored.

**Reply by icyJoseph** [reply]
Well, I don't know enough about it just yet, so I can't answer that question. I'll try to read up.

**Reply by umang0620** [reply]
Thanks @icyJoseph. If you find anything, please let me know , I’ll also check on my side.
